resources/views/docs/1/digging-deeper/child-process.md

```diff
@@ -0,0 +1,300 @@
+is always running. This works similarly to tools like [`supervisord`](http://supervisord.org/), ensuring that the
+process gets booted up again in case it crashes.
+
+```php
+ChildProcess::start(
+    cmd: ['tail', '-n50', 'logs/laravel.log'],
+    alias: 'tail',
+    persistent: true
+);
+```
+
+**The only way to stop a persistent process is for your application to quit.**
+
+## Getting running processes
+
+### Getting a single process
+You can use the `ChildProcess` facade's `get` method to get a running process with a given alias:
+
+```php
+$tail = ChildProcess::get('tail');
+```
+
+This will return a `Native\Laravel\ChildProcess` instance.
+
+### Getting all processes
+You can use the `ChildProcess` facade's `all` method to get all running processes:
+
+```php
+$tail = ChildProcess::all();
+```
+
+This will return an array of `Native\Laravel\ChildProcess` instances.
+
+## Sending input
+There are multiple ways to provide input to your Child Process:
+
+- The environment.
+- Arguments to the command.
+- Its standard input stream (`STDIN`).
+- A custom interface, e.g. a network socket.
+
+Which you use will depend on what the program is capable of handling.
+
+### Environment
+Child Processes will inherit the environment available to your application by default. If needed, you can provide extra
+environment variables when starting the process via the `$env` parameter of the `start` method:
+
+```php
+ChildProcess::start(
+    cmd: 'tail ...',
+    alias: 'tail',
+    env: [
+        'CUSTOM_ENV_VAR' => 'custom value',
+    ]
+);
+```
+
+### Command line arguments
+You can pass arguments to the program via the `$cmd` parameter of the `start` method. This accepts a `string` or an
+`array`, whichever you prefer to use:
+
+```php
+ChildProcess::start(
+    cmd: ['tail', '-n50', 'storage/logs/laravel.log'],
+    alias: 'tail'
+);
+```
+
+### Messaging a Child Process
+You may send messages to a running child process's standard input stream (`STDIN`) using the `message` method:
+
+```php
+$tail->message('Hello, world!');
+```
+
+Alternatively, you may use the `ChildProcess` facade to message a process via its alias:
+
+```php
+ChildProcess::message('Hello, world!', 'tail');
+```
+
+The message format and how they are handled will be determined by the program you're running.
+
+## Handling output
+A Child Process may send output via any of the following interfaces:
+
+- Its standard output stream (`STDOUT`).
+- Its standard error stream (`STDERR`).
+- A custom interface, e.g. a network socket.
+
+### Listening for Output (`STDOUT`)
+You may receive standard output for a process by registering an event listener for the
+[`MessageReceived`](#codemessagereceivedcode) event:
+
+```php
+use Illuminate\Support\Facades\Event;
+use Native\Laravel\Events\ChildProcess\MessageReceived;
+
+Event::listen(MessageReceived::class, function (MessageReceived $event) {
+    if ($event->alias === 'tail') {
+        info($event->data);
```

Comment:
If I'm not mistaking this only works from the service provider boot method. The example could be mistaken to work if put somewhere else, like a controller.

In the Events section below there is another way to register listeners.

These chapters serve a similar purpose and somewhat describe the same thing. There are two other sections in between (stopping & restarting). There might be an opportunity here to add to the overall cohesion of this doc.

Reply:
addressed in f1b9a94112cac3885d1b319a52bb148c57f29295. Thoughts @simonhamp?